client/landing/stepper/declarative-flow/internals/steps-repository/playground/lib/initialize-playground.ts

```diff
@@ -11,16 +11,23 @@ const OPFS_PATH_PREFIX = '/wpcom-onboarding';
 
 export async function initializeWordPressPlayground(
 	iframe: HTMLIFrameElement,
-	recommendedPhpVersion: string
+	recommendedPhpVersion: string,
+	setSearchParams: ( callback: ( prev: URLSearchParams ) => URLSearchParams ) => void
 ): Promise< PlaygroundClient > {
 	let isWordPressInstalled = false;
 
 	const url = new URL( window.location.href );
 	let playgroundId = url.searchParams.get( 'playground' );
 	if ( ! playgroundId ) {
 		playgroundId = crypto.randomUUID();
+		// update url in browser history
 		url.searchParams.set( 'playground', playgroundId );
 		window.history.replaceState( {}, '', url.toString() );
```

Comment:
[`setSearchParams` updates the navigation,](https://api.reactrouter.com/v7/types/react_router.SetURLSearchParams.html) so we don't need these lines anymore.

Reply:
No, I tried it locally and it worked for me, but if you have the issue let's leave it as-is.

Reply:
I don't have experience with react-router-dom, but from reading their docs it sounded like the navigation should be updated.

Reply:
@bgrgicak Just to confirm how I tested this: Without these lines, if I am going back by the browser's back button, I get to `http://calypso.localhost:3000/setup/onboarding/playground?playground=9a566c25-a356-4b43-8438-ccfc6fc3cbf0` and then `http://calypso.localhost:3000/setup/onboarding/playground` so history state wasn't replaced.